src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -114,11 +114,15 @@ export class PtyService extends Disposable implements IPtyService {
 		this._throwIfNoPty(id).detach();
 	}
 
+	reduceConnectionGraceTime(): void {
+		for (const pty of this._ptys.values()) {
+			pty.reduceGraceTime();
+		}
+	}
+
 	async listProcesses(reduceGraceTime: boolean): Promise<IProcessDetails[]> {
 		if (reduceGraceTime) {
-			for (const pty of this._ptys.values()) {
-				pty.reduceGraceTime();
-			}
+			this.reduceConnectionGraceTime();
 		}
```

Comment:
I think we want to remove this capability from `listProcesses` now since we have the explicit call